crates/cheatnet/src/execution/contract_syscall_handler.rs

```diff
@@ -0,0 +1,82 @@
+use cairo_felt::Felt252;
+use cairo_lang_casm::{
+    hints::{Hint, StarknetHint},
+    operand::ResOperand,
+};
+use cairo_lang_runner::{
+    casm_run::{extract_relocatable, vm_get_range},
+    short_string::as_cairo_short_string,
+};
+use cairo_vm::hint_processor::hint_processor_definition::HintProcessorLogic;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::vm::{errors::hint_errors::HintError, vm_core::VirtualMachine};
+use std::any::Any;
+use std::collections::HashMap;
+
+fn extract_input(
+    vm: &mut VirtualMachine,
+    input_start: &ResOperand,
+    input_end: &ResOperand,
+) -> Result<Vec<Felt252>, HintError> {
+    let input_start = extract_relocatable(vm, input_start)?;
+    let input_end = extract_relocatable(vm, input_end)?;
+    vm_get_range(vm, input_start, input_end)
+        .map_err(|_| HintError::CustomHint("Failed to read input data".into()))
+}
+
+pub(crate) struct ContractSyscallHandler<'a> {
```

Comment:
```suggestion
pub(crate) struct ContractExecutionSyscallHandler<'a> {
```
For the consistency with `TestExecutionSyscallHandler`

Reply:
[Address review suggestions](https://github.com/foundry-rs/starknet-foundry/pull/924/commits/a9245c7ef18afb28f9dee6d75699d9b040e7b802)